### PR TITLE
Damnation

### PR DESCRIPTION
**Pull request by wang429**

### CODE REVIEW

File: src/main/java/stsjorbsmod/powers/DamnationPower.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        updateDescription();
+    }
+
+    @Override
+    public void onUseCard(AbstractCard card, UseCardAction action) {
+        CardMetaUtils.destroyCardPermanently(card);
+        AbstractDungeon.transformCard(card, isAutoUpgrade, AbstractDungeon.miscRng);
+        AbstractDungeon.topLevelEffectsQueue.add(new ShowCardAndObtainEffect(AbstractDungeon.getTransformedCard(), Settings.WIDTH / 2.0F, Settings.HEIGHT / 2.0F, false));
+        addToBot(new RemoveSpecificPowerAction(owner, owner, this.ID));
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/powers/DamnationPower.java]: This should probably add to top; if there's a queue of card uses queued somehow, we don't want the power applying to several of them

File: src/main/java/stsjorbsmod/powers/DamnationPower.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        super(STATIC);
+
+        this.owner = owner;
+        this.isAutoUpgrade = isAutoUpgrade;
+
+        updateDescription();
+    }
+
+    @Override
+    public void onUseCard(AbstractCard card, UseCardAction action) {
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/powers/DamnationPower.java]: Should we exclude cards that aren't in the master deck? If not, should we at least exclude transient copies of cards used by "play next card twice" type mechanics? (Probably need to ask discord)

File: src/main/resources/stsjorbsmodResources/localization/eng/JorbsMod-Power-Strings.json [hunk 3]
```diff
@@ -46,6 +46,14 @@
       "Whenever you remember a new memory, deal #b%1$s damage to ALL enemies and lose all Coil."
     ]
   },
+  "stsjorbsmod:DamnationPower": {
+    "NAME": "Damnation",
+    "DESCRIPTIONS": [
+      "Permanently #yTransform the next card played this combat",
```
Comment (dbjorge) [inline src/main/resources/stsjorbsmodResources/localization/eng/JorbsMod-Power-Strings.json]: It's more localization friendly to just have 2 versions of the whole sentence written out, like in the card strings; it's okay for it to be a little redundant, it makes it easier for languages where the more natural phrasing might put the then clause before the object of the sentence

File: src/main/java/stsjorbsmod/powers/DamnationPower.java (hunk 2 above)
Comment (wang429) [reply src/main/java/stsjorbsmod/powers/DamnationPower.java]: Still asking about cards that aren't in the master deck

Comment (wang429) [reply src/main/java/stsjorbsmod/powers/DamnationPower.java]: cards not in the master deck will consume the charge without transforming

File: src/main/java/stsjorbsmod/powers/DamnationPower.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            CardMetaUtils.destroyCardPermanently(card);
+            AbstractDungeon.transformCard(card, isAutoUpgrade, AbstractDungeon.miscRng);
+            AbstractDungeon.topLevelEffectsQueue.add(new ShowCardAndObtainEffect(AbstractDungeon.getTransformedCard(), Settings.WIDTH / 2.0F, Settings.HEIGHT / 2.0F, false));
+        }
+        addToTop(new RemoveSpecificPowerAction(owner, owner, this.ID));
+    }
+
+    @Override
+    public void updateDescription() {
+        description = isAutoUpgrade ? DESCRIPTIONS[0] : DESCRIPTIONS[1];
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/powers/DamnationPower.java]: I think this is inverted from how the power strings are ordered, no?

**Pull request merged**